prov/sockets/src/sock_conn.c

```diff
@@ -423,7 +428,12 @@ struct sock_conn *sock_ep_connect(struct sock_ep_attr *ep_attr, fi_addr_t index)
 
 	ret = connect(conn_fd, (struct sockaddr *) addr, sizeof *addr);
 	if (ret < 0) {
+#ifndef _WIN32
 		if (errno == EINPROGRESS) {
+#else
+		int wsaerror = WSAGetLastError();
+		if(wsaerror == WSAEINPROGRESS || wsaerror == WSAEWOULDBLOCK) {
+#endif
```

Comment:
gurgle... maybe we define sockerrno, then define it to errno on non-windows platforms, and on windows to a macro that calls WSAGetLastError and converts the WSA error numbers to linux errno values.